homeassistant/components/switch/mqtt.py

```diff
@@ -114,6 +115,13 @@ def state_message_received(topic, payload, qos):
                 self.hass, self._state_topic, state_message_received,
                 self._qos)
 
+        if self._optimistic:
+            last_state = yield from async_get_last_state(self.hass,
+                                                         self.entity_id)
+            if last_state:
+                self._state = last_state.state
+                self.async_schedule_update_ha_state()
```

Comment:
No need to schedule an update. We write the entity to the state machine right after this method is done.